lib/service/connect.go

```diff
@@ -684,6 +688,115 @@ func (process *TeleportProcess) firstTimeConnect(role types.SystemRole) (*Connec
 	return connector, nil
 }
 
+func (process *TeleportProcess) initOpenSSH() {
+	process.RegisterWithAuthServer(types.RoleNode, SSHIdentityEvent)
+	process.SSHd = openssh.NewSSHD()
+	process.RegisterCriticalFunc("common.rotate_openssh", process.syncOpenSSHRotationState)
+}
+
+func (process *TeleportProcess) syncOpenSSHRotationState() error {
+	if _, err := process.WaitForEvent(process.GracefulExitContext(), TeleportReadyEvent); err != nil {
+		return nil
+	}
+	conn, err := process.WaitForConnector(SSHIdentityEvent, nil)
+	if conn == nil {
+		return trace.Wrap(err)
+	}
+	defer conn.Close()
+
+	ctx := context.Background()
+
+	status, err := process.syncRotationStateAndBroadcast(conn)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	fmt.Println(status)
+
+	id, err := process.storage.ReadIdentity(auth.IdentityCurrent, types.RoleInstance)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	cas, err := conn.Client.GetCertAuthorities(ctx, types.OpenSSHCA, false)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	keysDir := process.Config.OpenSSH.KeysDir
+	if err := openssh.TryCreateDefaultAgentlesKeysDir(keysDir); err != nil {
```

Comment:
I think that `openssh.WriteKeys` could just try to create the directory instead of requiring the caller to do this extra step.